src/database/connection.py

```
@@ -52,6 +69,127 @@ def __init__(self, minconn: int, maxconn: int, **conn_kwargs: Any):
+
+        Returns:
+            bool: True if health check passes, False otherwise.
+        """
+        try:
+            with self._pool.connection() as conn:
+                # Simple query to test connection health
+                with conn.cursor() as cur:
+                    cur.execute("SELECT 1 as health_check")
+                    result = cur.fetchone()
+                    return result is not None and result[0] == 1
+        except Exception as exc:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)